### ISSUE TITLE
Notifications in NC do not get propagated to android

### ISSUE THREAD

**Issue by christianlupus** [nextcloud/android#5129]
### Steps to reproduce
1. Set up the android app.
2. Wait or generate a few notification messages in the NC server by another user
3. Look at the android notification list (pulling down from the upper border)

### Expected behaviour
- The notifications in the server (e.g. user X has changed a calendar entry) should be visible soon after the message appears in the web frontend.

### Actual behaviour
- On one phone I have access (my girl-friend's one) this works just as expected
- In my phone I do not get any notifications from the app regarding NC notifications. Until recently, I was not aware of this feature at all as it was never shown to me.

### Environment data
Android version: 9

Device model: Xiaomi Mi A2 Lite

Stock or customized system: stock

Nextcloud app version: 3.9.2

Nextcloud server version: 7.0.1

### Logs
I doubt there is a problem on the server side. I thus omit the logs here. If you need them, please tell me what logs you need (during which period of time), so I can provide them.

### Additional information
Could this be related to the locking of the app?

**Comment by tobiasKaminsky** [comment]
What do you see if you go manually to notifications?

**Comment by christianlupus** [comment]
You mean in the app? Then I see a clone of the notifications I have in the NC web interface.

**Comment by AndyScherzinger** [comment]
@christianlupus do both phones use the GPlay version? (since the f-droid version doesn't support push notifications)

**Comment by christianlupus** [comment]
> @christianlupus do both phones use the GPlay version? (since the f-droid version doesn't support push notifications)

I'd say so. I installed once the dev version of nextcloud in parallel but removed it in the meantime (see #3421). At least on my phone (non-functional) I used the Play Store to install the app.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky any ideas?

**Comment by christianlupus** [comment]
Just out of curiosity: If the app is protected by pin/credentials, it would be obvious to me (by my intention), that there are no notifications shown to avoid leaking any information. Is this implemented that way?

**Comment by AndyScherzinger** [comment]
> Is this implemented that way?

Not that I am aware of but I pinged @tobiasKaminsky since he implemented that part f the app and should know best.

**Comment by tobiasKaminsky** [comment]
> You mean in the app? Then I see a clone of the notifications I have in the NC web interface.

Reason I asked is that there should be a snackbar about failing push notifications.



> > Is this implemented that way?
> 
> Not that I am aware of but I pinged @tobiasKaminsky since he implemented that part f the app and should know best.

Our app obeys the system wide notification-privacy settings, and not if app protection is enabled (as this is only to prevent access)

**Comment by christianlupus** [comment]
> > You mean in the app? Then I see a clone of the notifications I have in the NC web interface.
> 
> Reason I asked is that there should be a snackbar about failing push notifications.

Nope, I do not get a bunch of these.

> > > Is this implemented that way?
> > 
> > 
> > Not that I am aware of but I pinged @tobiasKaminsky since he implemented that part f the app and should know best.
> 
> Our app obeys the system wide notification-privacy settings, and not if app protection is enabled (as this is only to prevent access)

OK, the system allows the app to show notificatios for sure (I get the message, if a new file is to be auto-uploaded etc). So this is not the problem. I have the privacy protection enabled but this should only  affect the notifications in case the system is not locked.

**Comment by msilvoli** [comment]
> > You mean in the app? Then I see a clone of the notifications I have in the NC web interface.
> 
> Reason I asked is that there should be a snackbar about failing push notifications.
> 
> > > Is this implemented that way?
> > 
> > 
> > Not that I am aware of but I pinged @tobiasKaminsky since he implemented that part f the app and should know best.
> 
> Our app obeys the system wide notification-privacy settings, and not if app protection is enabled (as this is only to prevent access)

In my case I don't see nothing going manually in app notifications. When I send admin notifications to users I can't see anything in notification app as well. Android not receiving notifications from the server in the app.
Server: 17.0.2.1
App: 3.10 (Gplay) no battery restrictions, notifications enabled (including locked screen)
Android 9 (MIUI 11)

**Comment by christianlupus** [comment]
> Android 9 (MIUI 11)
Interesting, we both have a Xiaomi phone although I have a Android one (aka clean Android) instead of a MIUI.

**Comment by Chartman123** [comment]
What happens when you clear the notification within the app? If it's crashing it could be related to another issue... #4586

**Comment by christianlupus** [comment]
OK, I tried a few things again.

First, when I click on the cross in the app next to a notification, this notification is removed successfully.

Second, when I open a browser on my laptop and the android app on the notification's view, I see similar views. I keep the android phone on but avoid manual sync (I do not pull down the notifications to get the running circle). Now I create a notification using `php occ notification:generate`. On the browser it gets visible 5-10 sec after I issued the command. In the app it does not get visible (unless I manually sync). I did not wait multiple minutes at some point there might be some polling in place.

**Comment by AndyScherzinger** [comment]
cc @tobiasKaminsky - my guess something is off with the push serve or the client's registration for it.

**Comment by msilvoli** [comment]
What I 

> What happens when you clear the notification within the app? If it's crashing it could be related to another issue... #4586

What I'm experiencing:
In the beggining of the problem notifications were not received by push, but they showed up there inside notifications in the App when I refreshed the page manually - it doesn't matter if it was admin notification or user notification. This situation stayed for a while until 3 days ago when you asked me about the issue #4586.
I left the App opened in the notification screen (there were no notifications there as usual) and I sent an administrator announcement to my test user. I didn't have time to see anything than the crash mentioned by  #4586. I didn't touch anything, it crashed without any interaction so I think that in someway the app received the notification. After this occurence (crash) nothing more is arriving to my phone - automatically or refreshing the notification area manually. Restarted apache server but nothing happened. Everything related to notification stopped in the app, but I think the server is sending because I still receive the e-mail notification.

Have a look in the error log when it crashed (a few extra lines with file path to the client):

************ CAUSE OF ERROR ************

java.lang.NoClassDefFoundError: Failed resolution of: Lcom/google/firebase/analytics/connector/AnalyticsConnector;
	at com.google.firebase.messaging.MessagingAnalytics.zza(Unknown Source:68)
	at com.google.firebase.messaging.MessagingAnalytics.logNotificationReceived(Unknown Source:2)
	at com.google.firebase.messaging.FirebaseMessagingService.zzd(Unknown Source:50)
	at com.google.firebase.iid.zzg.run(Unknown Source:4)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at com.google.android.gms.common.util.concurrent.zza.run(Unknown Source:7)
	at java.lang.Thread.run(Thread.java:764)
Caused by: java.lang.ClassNotFoundException: Didn't find class "com.google.firebase.analytics.connector.AnalyticsConnector" on path: DexPathList[[zip file "/system/framework/org.apache.http.legacy.boot.jar", zip file "/data/app/com.nextcloud.client-HFAwewHFVsluT0uTf9FyYw==/base.apk"],nativeLibraryDirectories=[/data/app/com.nextcloud.client-HFAwewHFVsluT0uTf9FyYw==/lib/arm64, /data/app/com.nextcloud.client-HFAwewHFVsluT0uTf9FyYw==/base.apk!/lib/arm64-v8a, /system/lib64]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:171)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
	... 8 more

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30100090
Build flavor: gplay

************ DEVICE INFORMATION ************
Brand: Xiaomi
Device: davinci
Model: Mi 9T
Id: PKQ1.190302.001
Product: davinci_eea

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: V11.0.3.0.PFJEUXM

Server: 17.0.2.1 / Apache: 2.4 / MariaDB: 10.2.29 / PHP 7.2.5 Opcache / Redis
No encryption on server side / No TOTP
App: 3.10 (Gplay) no battery restrictions, notifications enabled (including locked screen)
Android 9 (MIUI 11) Global Version

**Comment by AndyScherzinger** [comment]
@msilvoli - your error is a duplicate of #4675

**Comment by christianlupus** [comment]
Is there any chance/plan that in case the push is not working (i.e. non-Google-Play installation), that a polling approach is used to fetch the notifications?

**Comment by Valdnet** [comment]
@AndyScherzinger @tobiasKaminsky In my case on Android 9 I also don't always get push notifications. I get them only after starting the Nextcloud Android application.

On Android 8, I get them immediately.

**Comment by tobiasKaminsky** [comment]
Duplicate of #4675

**Comment by christianlupus** [comment]
Are you sure? #4675 is referring to an exception been thrown. I have never seen any `NoClassDefFoundException` while using the app or during usage of the phone. Or is it somewhere in the logs visible only?

**Comment by tobiasKaminsky** [comment]
I am referring to this: https://github.com/nextcloud/android/issues/5129#issuecomment-579347040

@christianlupus so you do not see any notification in notification view?

**Comment by christianlupus** [comment]
You mean in the app? Yes, there are the notifications visible that are visible in the web frontend as well.

**Comment by christianlupus** [comment]
I just tried it as suggested by the **msilvoli**: Open the app with the notifications open. There was an update notification still present. I issued an admin notification using `occ notification:generate`.
It did **not** appear in the app immediately nor did the app break/freeze/explode. I then confirmed the notification in the web frontend (simply installed the update). Still the old message was shown in the app and the new message from `occ` was not showing. After a minute or so I force-reloaded by pulling down in the app. A second later, the old notification was gone and the new onw present.

**Comment by tobiasKaminsky** [comment]
Notification view in app is not updated dynamically, but only when entering or when force-reloading.

So this is really related to push, but I do not know why.
Do you get other push notifications? E.g. from your bank app, etc?

**Comment by christianlupus** [comment]
Yes, I do. Other apps seem to work fine.

**Comment by msilvoli** [comment]
Hello, 
I'm still struggling with notifications. Sometimes it appears on the Nextcloud APP and sometimes not. Changes a lot of config on phone but nothing comes.
Any other suggestions ?

**Comment by christianlupus** [comment]
I still have the issue.

**Comment by christianlupus** [comment]
Unfortunately, this issue is not closed the stale label, although I added a comment.

In the meantime, I updated to Android 10 and the issue is still present. So, I ask for the removal of the stale label as the topic is still valid. How can I help getting things rolling?

**Comment by christianlupus** [comment]
I am still unable to have this running on my device. Neither talk nor normal notifications are delivered. Running on 3.12.1 installation of the main app at the moment.
Is there a way to debug the app in any way to see what's going on? Or writing a trace/log file once a certain flag file exists?

**Comment by AndyScherzinger** [comment]
> Is there a way to debug the app in any way to see what's going on? Or writing a trace/log file once a certain flag file exists?

This depends on the version. In case of the DEV versions as well as the Beta releases (RC*) there is a log available via the settings screen.

Other than that maybe @tobiasKaminsky has an idea. My guess would be that the push registration didn't happen (or is broken somehow) so there won't be much too see I guess while Tobias might be able so shed some light onto this since there might be some detection/healing during app starts.

**Comment by christianlupus** [comment]
But the DEV version is not registered with the google play store thus rendering notifications non-functional. I am on the beta program of the Nextcloud app. I am unsure if the beta app is installed as I think I did not see any logging functionality in the settings.

**Comment by Valdnet** [comment]
> But the DEV version is not registered with the google play store thus rendering notifications non-functional.

Exactly. I am also missing PUSH notifications in the DEV version to investigate possible problems with notifications.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky probably knows best.

**Comment by tobiasKaminsky** [comment]
Duplicate of #4675

**Comment by christianlupus** [comment]
Hello @tobiasKaminsky, I have not worked my way through the whole bunch of comments in the other thread. But they seem to have valid error messages. I do not get any sort of error messages which is one of the main culprits of this issue as our is hard to reproduce. 
So: are you sure this thread is a duplicate? Where might I get such an error message?

**Comment by tobiasKaminsky** [comment]
Via logcat: https://github.com/nextcloud/android/blob/master/README.md#getting-debug-info-via-logcat
